### PR TITLE
Add theme switching functionality using Teams sdk

### PR DESCRIPTION
**Pull request by FaizChishtie**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -29,11 +35,18 @@ class App extends React.Component<{}, IAppState>{
     if(viewOption !== undefined){
       this.setState({viewOption: viewOption});
     }
-    console.log(this.state);
+  }
+
+  public componentDidMount() {
+    microsoftTeams.initialize();
+    microsoftTeams.getContext((context: microsoftTeams.Context) => {
```
Comment (sachingupta) [inline src/App.tsx]: remove this one. we are not using it now.

File: src/ProviderWrapper.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import App from './App';
+
+enum themeTypes{
+    Dark='dark',
+    Default='default',
+    Contrast='contrast'
+};
+
+// wraps provider component so that theme can be held in state
+export class ProviderWrapper extends React.Component<any, any>{
```
Comment (sachingupta) [inline src/ProviderWrapper.tsx]: define interface for props and state of this component

File: src/ProviderWrapper.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    render(){
+        return (
+            <Provider theme={this.state.currentTheme}> 
+                <App onThemeChange={this.handleThemeChange}/>
+            </Provider>
+        );
+    }
+
+    // gets theme name from url params
+    public getThemeFromURL(url: string): string{
```
Comment (sachingupta) [inline src/ProviderWrapper.tsx]: move these functions to utils.ts

File: src/index.tsx [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
+import { themes } from '@stardust-ui/react';
 
+import { ProviderWrapper } from './ProviderWrapper';
 
-ReactDOM.render(<Provider theme={themes.teams}>
-    <App />
-   </Provider>, document.getElementById('root'));
+
+
+ReactDOM.render(<ProviderWrapper currentTheme={themes.teams} />, document.getElementById('root'));
```
Comment (sachingupta) [inline src/index.tsx]: r u using it ? if not then remove this prop

File: src/ProviderWrapper.tsx (hunk 2 above)
Comment (FaizChishtie) [reply src/ProviderWrapper.tsx]: Removed props since none were being used

File: src/index.tsx (hunk 4 above)
Comment (FaizChishtie) [reply src/index.tsx]: removed

File: src/utils/themeUtils.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    var subs = url.substring(indexOfEquals+1);
+
+    return subs;
+}
+
+// gets theme type from string
+export const getTheme = (theme:string):ThemeInput => {
+    var newTheme:ThemeInput = themes.teams;
+    
+    if(theme === themeTypes.Contrast){
```
Comment (sachingupta) [inline src/utils/themeUtils.ts]: use a switch statement

File: src/ProviderWrapper.tsx (hunk 3 above)
Comment (sachingupta) [inline src/ProviderWrapper.tsx]: use https://www.npmjs.com/package/query-string

**Pull request merged**